File: 2 sum problem original.py
import numpy as np

class Solution:
	def twoSum(self,nums, target):
		ans = np.empty((2),dtype='int32')
		dic1 = {}
		for i in range(len(nums)):
			if (target - nums[i]) in dic1:
				ans[0] = dic1[target - nums[i]]
				ans[1] = i
				return ans
			dic1[nums[i]] = i
		return ans

if __name__ == '__main__':
	nums = np.array([2,6,5,8,11])
	target = 14
	obj = Solution()
	ans = obj.twoSum(nums,target)
	print(ans)

File: 2 sum problem - original.cpp

#include <iostream>
#include <unordered_map>
#include <vector>

using namespace std;


// Optimized solution - TC - O(N), SC - O(N)
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target){
        vector<int> ans;
        unordered_map<int,int> mpp;
        for(int i=0; i<nums.size();i++){
            if (mpp.find(target- nums[i]) != mpp.end() ) {
                ans.push_back(mpp[target-nums[i]]);
                ans.push_back(i);
                return ans;
            }
            mpp[nums[i]] = i;
        }
        return ans;
    }
};

int main(){
    vector<int> num{2,6,5,8,11};
    int target = 14;
    Solution obj;
    vector<int> ans = obj.twoSum(num, target);
    cout << ans[0]<<" "<< ans[1];
    return 0;
}
